Mathlib/Data/Sum/Basic.lean

```diff
@@ -111,6 +111,22 @@ theorem update_inr_apply_inr [DecidableEq β] [DecidableEq (α ⊕ β)] {f : α
     {x : γ} : update f (inr i) x (inr j) = update (f ∘ inr) i x j := by
   rw [← update_inr_comp_inr, Function.comp_apply]
 
+@[simp]
+theorem update_inl_comp_inl_apply [DecidableEq α] [DecidableEq (α ⊕ β)] {γ : α ⊕ β → Type*}
+    {f : (i : α ⊕ β) → γ i} {i : α}
+    {x : γ (.inl i)} (j : α) :
+    update f (.inl i) x (Sum.inl j) =
+      update (fun j ↦ f (.inl j)) i x j :=
```

Comment:
Given the name and similarity with `update_comp_eq_of_injective`', I'd expect this to be stated as
```suggestion
    (fun j => update f (.inl i) x (Sum.inl j) =
      update (fun j ↦ f (.inl j)) i x :=
```
I'd also expect the name to be primed, again for similarity.

Alternatively, let's introduce  an applied version of `Function.update_comp_eq_of_injective'`, which can be unprimed and match a theorem here. Maybe it can even be `simp`, which might make this theorem unnecessary.

Reply:
I have added the general `Function.update_comp_apply_of_injective`, but it cannot be made a `simp` lemma because it depends on a `Function.Injective` assumption.

Reply:
That doesn't necessarily mean it can't be `simp`, but it's reasonable to decide not to make it `simp` for now to be conservative.